### PR TITLE
Instantiate empty module

### LINKED ISSUE #636: Failure to instantiate valid empty Wasm module
Wasmtime fails to instantiate the valid, empty module:

```wat
(module)
```

Here is a Rust unit test:

```rust
#[test]
fn instantiate_empty_module() {
    // `(module)`
    let wasm = vec![0x0, 0x61, 0x73, 0x6d 0x01, 0x0, 0x0, 0x0];
    wasmtime_fuzzing::oracles::instantiate(&wasm, wasmtime_jit::CompilationStrategy::Auto);
}
```

### PR DESCRIPTION
**Pull request by data-pup**
Fixes #636.

This allows wasmtime to instantiate the valid, empty wasm module `(module)`.

While working on this, I found that adding said empty module to `misc_testsuite` would not cause a test failure, so I added a standard unit test. Right now that test lives in `tests/foo.rs` because I was not sure where to put it.

It _feels_ like this should live inside of a subdirectory, but I couldn't decide what to call it, given that there is a `misc_testsuite` already. Adding a `test/instantiation_testsuite` directory also felt deceptive, as this would not relate to `tests/instantiate.rs`. Let me know where to put that, and I can update this PR. And thanks to @fitzgen for including a prewritten test in the issue description :pray:

### CODE REVIEW

File: tests/foo.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+use cranelift_codegen::settings;
```
Comment (peterhuene) [inline tests/foo.rs]: If we do keep it as a unit test, perhaps this should be `instantiate_empty_module.rs` instead of `foo.rs`?

Comment (peterhuene) [reply tests/foo.rs]: Oh I see this was an active question above as where this should go, hence the name.  👍

**Comment by fitzgen** [comment]
> While working on this, I found that adding said empty module to `misc_testsuite` would not cause a test failure, so I added a standard unit test. Right now that test lives in `tests/foo.rs` because I was not sure where to put it.

I created the `crates/fuzzing/tests/regressions/` directory to hold WAT test cases for fuzzer-found regression tests, and the `crates/fuzzing/tests/regressions.rs` test file for the `#[test]`s that use those files. I think that would be a good location for this test.

**Comment by fitzgen** [comment]
That approach would also allow reusing `oracles::instantiate` without needing to introduce a dev-dependency on the `wasmtime-fuzzing` crate from the main crate.

**Comment by fitzgen** [comment]
(See https://github.com/bytecodealliance/wasmtime/pull/637/ for details on the introduction of the regressions test and directory)

**Comment by fitzgen** [comment]
Sorry, I wasn't clear in my previous comments. Here is how I'd like the regression test to be set up:

* the `(module)` WAT test case should go into `crates/fuzzing/tests/regressions/empty.wat`
* the `#[test]` that is currently in `instantiate_empty_module.rs` should go into `crates/fuzzing/tests/regressions.rs` as

  ```rust
  #[test]
  fn instantiate_empty_module() {
      let data = wat::parse_str(
          include_str!("./regressions/empty.wat")
      ).unwrap();
      oracles::instantiate(data, wasmtime_jit::CompilationStrategy::Auto);
  }
  ```

Apologies for forcing churn on you!

**Comment by data-pup** [comment]
No problem at all! There wasn't precedent to go off of, so it's totally natural to want this positioned correctly :slightly_smiling_face:  Fixed that, should be good to go now!

**Pull request merged**